From 56a1b2797b1eeb127ff8376b2d64331d7ab68fbf Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Thu, 18 May 2023 15:59:55 +0530
Subject: [PATCH] WA-Revert "util: Drop the ENUM_PACKED macro"

This reverts commit 13f7ef5e0c01cb6f39ad3147800338f3d5b1f13e.

Build Error-:
src/gallium/auxiliary/util/u_prim.c:33:4: error: static assertion failed
due to requirement.

size of packed enum was coming 4 instead of 1 due to this change.

Tracked-On: OAM-106853
Signed-off-by: Ankit Agrawal <ankit.agarwal@intel.com>
---
 src/gallium/include/pipe/p_defines.h | 4 ++--
 src/util/macros.h                    | 8 ++++++++
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/gallium/include/pipe/p_defines.h b/src/gallium/include/pipe/p_defines.h
index 9ba20522406..5953b56fa09 100644
--- a/src/gallium/include/pipe/p_defines.h
+++ b/src/gallium/include/pipe/p_defines.h
@@ -561,7 +561,7 @@ enum pipe_resource_usage {
 /**
  * Primitive types:
  */
-enum PACKED pipe_prim_type {
+enum pipe_prim_type {
    PIPE_PRIM_POINTS,
    PIPE_PRIM_LINES,
    PIPE_PRIM_LINE_LOOP,
@@ -578,7 +578,7 @@ enum PACKED pipe_prim_type {
    PIPE_PRIM_TRIANGLE_STRIP_ADJACENCY,
    PIPE_PRIM_PATCHES,
    PIPE_PRIM_MAX,
-};
+} ENUM_PACKED;
 
 /**
  * Tessellator spacing types
diff --git a/src/util/macros.h b/src/util/macros.h
index 84b2cddfbae..8e0ee0ae171 100644
--- a/src/util/macros.h
+++ b/src/util/macros.h
@@ -315,6 +315,14 @@ do {                       \
 #define ATTRIBUTE_NOINLINE
 #endif
 
+/* Use as: enum name { X, Y } ENUM_PACKED; */
+#if defined(__GNUC__)
+#define ENUM_PACKED __attribute__((packed))
+#else
+#define ENUM_PACKED
+#endif
+
+
 /**
  * Check that STRUCT::FIELD can hold MAXVAL.  We use a lot of bitfields
  * in Mesa/gallium.  We have to be sure they're of sufficient size to
-- 
2.17.1

